### PR TITLE
feat(Vouchers): Vouchers+System+Product: scan_prefixes, voucher_actio…

### PR DESCRIPTION
**Pull request by alghanor**
…ns, vouchers

Add functionality to reflect sub-systems

https://tillhub.atlassian.net/browse/TM-4876

### CODE REVIEW

File: lib/v0/vouchers/base.js [hunk 1]
```diff
@@ -218,5 +218,23 @@ module.exports = {
   is_campaign: {
     description: 'Mark this voucher to be a campaign itself. This should be used if you issue multiple vouchers with the same code and do not want to deduct value from it.',
     type: 'boolean'
+  },
+  system: {
```
Comment (eljefedelrodeodeljefe) [inline lib/v0/vouchers/base.js]: unfortunately the db key here is `voucher_system`

File: lib/v1/configurations/items/scan_prefixes.js [hunk 2]
```diff
@@ -10,25 +12,86 @@ module.exports = {
         properties: {
           id: {
             type: 'string',
-            maxLength: 64
+            format: 'uuid',
```
Comment (eljefedelrodeodeljefe) [inline lib/v1/configurations/items/scan_prefixes.js]: string

File: lib/v1/configurations/items/scan_prefixes.js [hunk 3]
```diff
@@ -10,25 +12,86 @@ module.exports = {
...
-            type: 'object'
+            type: 'object',
+            properties: {
+              id: {
+                type: 'string',
+                format: 'uuid',
+                description: 'The Tillhub system resource reference ID associated with this prefix.',
+                example: '43847a66-97dc-4ac2-8e8a-c44920e1f22f'
+              },
+              system: {
```
Comment (qtotuan) [inline lib/v1/configurations/items/scan_prefixes.js]: Is this what you meant when you told me about 'system.system'?Is it possible to rename this? Into "sub_system" or something like that?

File: lib/v1/configurations/items/voucher_actions.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+          system: {
+            type: 'object',
+            properties: {
+              id: {
+                type: 'string',
+                format: 'uuid',
+                description: 'The Tillhub system resource reference ID associated with this prefix.',
+                example: '43847a66-97dc-4ac2-8e8a-c44920e1f22f'
+              },
+              system: {
```
Comment (qtotuan) [inline lib/v1/configurations/items/voucher_actions.js]: Same as above. Is this actually referring to a "voucher_system"?

File: lib/v1/configurations/items/scan_prefixes.js (hunk 3 above)
Comment (alghanor) [reply lib/v1/configurations/items/scan_prefixes.js]: it reflects the current (in production) iOS implementation

File: lib/v1/configurations/items/voucher_actions.js (hunk 4 above)
Comment (alghanor) [reply lib/v1/configurations/items/voucher_actions.js]: it reflects the current (in production) iOS implementation

**Pull request merged**